semgrep/semgrep/core_runner.py

```diff
@@ -249,6 +253,77 @@ def execute(self) -> int:
         return rc
 
 
+@frozen
+class Task:
+    path: str = field(converter=str)
+    language: Language
+    rule_ids: Sequence[str]
+
+
+class Plan(List[Task]):
+    @property
+    def rule_count(self) -> int:
+        return len(set(rule for task in self for rule in task.rule_ids))
+
+    @property
+    def file_count(self) -> int:
+        return len(self)
+
+    def split_by_lang_label(self) -> Dict[str, "Plan"]:
+        result: Dict[str, Plan] = collections.defaultdict(Plan)
+        for task in self:
+            label = (
+                "<multilang>"
```

Comment:
@dlukeomalley's reasons against doing this:

> we are introducing language into our output that doesn’t match the underlying rules. I wonder if instead we accurately communicate the specified “language” of the rules running, so folks know that they have (perhaps less good) generic rules or are indeed running simple ol’ regex . I’m not strong on this one, just a flag that multilang is new. Perhaps generic would be better renamed to multilang to capture what it does :shrug:

my reasons for:

- I think the resolution between regex/generic is not important and merging can save a line of output; in typical usages such as our common recommendation of `p/ci + p/secrets` this'd often mean 2 lines instead of 3.
- regex and generic will always have the same file count which makes the extra line even less useful
- If we list "python, js, regex", that can be misinterpreted as "we're scanning this many of your regex patterns for correctness"
- `--config=auto` users will likely totally not care nor they should need to know what underlying tech is yielding results